### ISSUE TITLE
Driver is not removing Machines when used with gitlab-runner

### ISSUE THREAD

**Issue by AdamVB** [opentelekomcloud/docker-machine-opentelekomcloud#46]
Sometimes the vm id is not properly submitted during the delete request which then results in this error message:

```
Sep 14 10:05:22 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Problem while reading command output       error=read |0: file already closed
Sep 14 10:05:22 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Problem while reading command output       error=read |0: file already closed
Sep 14 10:05:22 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Stopping machine                           lifetime=10m20.060752882s name=runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a reason=Too many builds used=1m15.901210012s usedCount=1
Sep 14 10:05:22 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Stopping machine                           lifetime=10m20.060752882s name=runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a reason=Too many builds used=1m15.901210012s usedCount=1
Sep 14 10:05:22 gitlab-runner-bastion gitlab-runner[18655]: Stopping "runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a"...  name=runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a operation=stop
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: Stopping "runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a"...  name=runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a operation=stop
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: ERROR: Error creating machine: Error in driver during machine creation: Internal Server Error  driver=otc name=runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d operation=create
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: ERROR: Error creating machine: Error in driver during machine creation: Internal Server Error  driver=otc name=runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d operation=create
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Problem while reading command output       error=read |0: file already closed
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Problem while reading command output       error=read |0: file already closed
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Problem while reading command output       error=read |0: file already closed
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Problem while reading command output       error=read |0: file already closed
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Machine creation failed, trying to provision  error=exit status 1 name=runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Machine creation failed, trying to provision  error=exit status 1 name=runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Problem while reading command output       error=read |0: file already closed
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Problem while reading command output       error=read |0: file already closed
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: ERROR: Resource not found: [POST https://ecs.eu-de.otc.t-systems.com/v2.1/f63d1faef3f747e9aa5cbcf017252ed1/servers/f2658670-2253-47e2-9e31-e104a7fd0ed4/action], error message: {"itemNotFound": {"message": "Instance f2658670-2253-47e2-9e31-e104a7fd0ed4 could not be found.", "code": 404}}  name=runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a operation=stop
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: ERROR: Resource not found: [POST https://ecs.eu-de.otc.t-systems.com/v2.1/f63d1faef3f747e9aa5cbcf017252ed1/servers/f2658670-2253-47e2-9e31-e104a7fd0ed4/action], error message: {"itemNotFound": {"message": "Instance f2658670-2253-47e2-9e31-e104a7fd0ed4 could not be found.", "code": 404}}  name=runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a operation=stop
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Problem while reading command output       error=read |0: file already closed
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Problem while reading command output       error=read |0: file already closed
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Error while stopping machine               error=exit status 1 lifetime=10m20.902521558s name=runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a reason=Too many builds used=1m16.742978665s usedCount=1
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Error while stopping machine               error=exit status 1 lifetime=10m20.902521558s name=runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a reason=Too many builds used=1m16.742978665s usedCount=1
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Removing machine                           lifetime=10m20.903625852s name=runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a reason=Too many builds used=1m16.744082799s usedCount=1
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Removing machine                           lifetime=10m20.903625852s name=runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a reason=Too many builds used=1m16.744082799s usedCount=1
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: About to remove runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a  name=runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a operation=remove
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: About to remove runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a  name=runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a operation=remove
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: This action will delete both local reference and remote instance.  name=runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a operation=remove
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: This action will delete both local reference and remote instance.  name=runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a operation=remove
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: ERROR: Resource not found: [POST https://ecs.eu-de.otc.t-systems.com/v2.1/f63d1faef3f747e9aa5cbcf017252ed1/servers/84a4d5eb-dba8-462e-a900-a7107e397626/action], error message: {"itemNotFound": {"message": "Instance 84a4d5eb-dba8-462e-a900-a7107e397626 could not be found.", "code": 404}}  name=runner-kutriyer-gitlab-runner-autoscale-1600077312-1cfcec4b operation=stop
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: ERROR: Resource not found: [POST https://ecs.eu-de.otc.t-systems.com/v2.1/f63d1faef3f747e9aa5cbcf017252ed1/servers/84a4d5eb-dba8-462e-a900-a7107e397626/action], error message: {"itemNotFound": {"message": "Instance 84a4d5eb-dba8-462e-a900-a7107e397626 could not be found.", "code": 404}}  name=runner-kutriyer-gitlab-runner-autoscale-1600077312-1cfcec4b operation=stop
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Problem while reading command output       error=read |0: file already closed
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Problem while reading command output       error=read |0: file already closed
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Problem while reading command output       error=read |0: file already closed
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Problem while reading command output       error=read |0: file already closed
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Error while stopping machine               error=exit status 1 lifetime=10m10.916268843s name=runner-kutriyer-gitlab-runner-autoscale-1600077312-1cfcec4b reason=Too many builds used=1m17.584298623s usedCount=1
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Error while stopping machine               error=exit status 1 lifetime=10m10.916268843s name=runner-kutriyer-gitlab-runner-autoscale-1600077312-1cfcec4b reason=Too many builds used=1m17.584298623s usedCount=1
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Removing machine                           lifetime=10m10.917450343s name=runner-kutriyer-gitlab-runner-autoscale-1600077312-1cfcec4b reason=Too many builds used=1m17.585479957s usedCount=1
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Removing machine                           lifetime=10m10.917450343s name=runner-kutriyer-gitlab-runner-autoscale-1600077312-1cfcec4b reason=Too many builds used=1m17.585479957s usedCount=1
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: About to remove runner-kutriyer-gitlab-runner-autoscale-1600077312-1cfcec4b  name=runner-kutriyer-gitlab-runner-autoscale-1600077312-1cfcec4b operation=remove
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: About to remove runner-kutriyer-gitlab-runner-autoscale-1600077312-1cfcec4b  name=runner-kutriyer-gitlab-runner-autoscale-1600077312-1cfcec4b operation=remove
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: This action will delete both local reference and remote instance.  name=runner-kutriyer-gitlab-runner-autoscale-1600077312-1cfcec4b operation=remove
Sep 14 10:05:23 gitlab-runner-bastion gitlab-runner[18655]: WARNING: This action will delete both local reference and remote instance.  name=runner-kutriyer-gitlab-runner-autoscale-1600077312-1cfcec4b operation=remove
Sep 14 10:05:24 gitlab-runner-bastion gitlab-runner[18655]: ERROR: Error removing host "runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a": 1 error occurred:  name=runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a operation=remove
Sep 14 10:05:24 gitlab-runner-bastion gitlab-runner[18655]: ERROR: Error removing host "runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a": 1 error occurred:  name=runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a operation=remove
Sep 14 10:05:24 gitlab-runner-bastion gitlab-runner[18655]: ERROR:         * Resource not found: [DELETE https://ecs.eu-de.otc.t-systems.com/v2.1/f63d1faef3f747e9aa5cbcf017252ed1/servers/f2658670-2253-47e2-9e31-e104a7fd0ed4], error message: {"itemNotFound": {"message": "Instance f2658670-2253-47e2-9e31-e104a7fd0ed4 could not be found.", "code": 404}}  name=runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a operation=remove
Sep 14 10:05:24 gitlab-runner-bastion gitlab-runner[18655]: ERROR:         * Resource not found: [DELETE https://ecs.eu-de.otc.t-systems.com/v2.1/f63d1faef3f747e9aa5cbcf017252ed1/servers/f2658670-2253-47e2-9e31-e104a7fd0ed4], error message: {"itemNotFound": {"message": "Instance f2658670-2253-47e2-9e31-e104a7fd0ed4 could not be found.", "code": 404}}  name=runner-kutriyer-gitlab-runner-autoscale-1600077302-0518922a operation=remove


[...}


Sep 14 10:23:09 gitlab-runner-bastion gitlab-runner[18655]: Stopping "runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d"...  name=runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d operation=stop
Sep 14 10:23:09 gitlab-runner-bastion gitlab-runner[18655]: ERROR: Resource not found: [POST https://ecs.eu-de.otc.t-systems.com/v2.1/f63d1faef3f747e9aa5cbcf017252ed1/servers//action], error message: {"message":"API not found","request_id":"09557bbd78e2afafdd49359271110e9a"}  name=runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d operation=stop
Sep 14 10:23:09 gitlab-runner-bastion gitlab-runner[18655]: ERROR: Resource not found: [POST https://ecs.eu-de.otc.t-systems.com/v2.1/f63d1faef3f747e9aa5cbcf017252ed1/servers//action], error message: {"message":"API not found","request_id":"09557bbd78e2afafdd49359271110e9a"}  name=runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d operation=stop
Sep 14 10:23:09 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Problem while reading command output       error=read |0: file already closed
Sep 14 10:23:09 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Problem while reading command output       error=read |0: file already closed
Sep 14 10:23:09 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Problem while reading command output       error=read |0: file already closed
Sep 14 10:23:09 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Problem while reading command output       error=read |0: file already closed
Sep 14 10:23:09 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Error while stopping machine               error=exit status 1 lifetime=17m57.022048657s name=runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d reason=Failed to create used=8m42.669057402s usedCount=0
Sep 14 10:23:09 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Error while stopping machine               error=exit status 1 lifetime=17m57.022048657s name=runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d reason=Failed to create used=8m42.669057402s usedCount=0
Sep 14 10:23:09 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Removing machine                           lifetime=17m57.023203801s name=runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d reason=Failed to create used=8m42.670212051s usedCount=0
Sep 14 10:23:09 gitlab-runner-bastion gitlab-runner[18655]: WARNING: Removing machine                           lifetime=17m57.023203801s name=runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d reason=Failed to create used=8m42.670212051s usedCount=0
Sep 14 10:23:09 gitlab-runner-bastion gitlab-runner[18655]: About to remove runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d  name=runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d operation=remove
Sep 14 10:23:09 gitlab-runner-bastion gitlab-runner[18655]: About to remove runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d  name=runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d operation=remove
Sep 14 10:23:09 gitlab-runner-bastion gitlab-runner[18655]: WARNING: This action will delete both local reference and remote instance.  name=runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d operation=remove
Sep 14 10:23:09 gitlab-runner-bastion gitlab-runner[18655]: WARNING: This action will delete both local reference and remote instance.  name=runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d operation=remove
Sep 14 10:23:09 gitlab-runner-bastion gitlab-runner[18655]: ERROR: Error removing host "runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d": 1 error occurred:  name=runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d operation=remove
Sep 14 10:23:09 gitlab-runner-bastion gitlab-runner[18655]: ERROR: Error removing host "runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d": 1 error occurred:  name=runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d operation=remove
Sep 14 10:23:09 gitlab-runner-bastion gitlab-runner[18655]: ERROR:         * Resource not found: [DELETE https://ecs.eu-de.otc.t-systems.com/v2.1/f63d1faef3f747e9aa5cbcf017252ed1/servers/], error message: {"message":"API not found","request_id":"3ff1331e9c21dec814772ea059475c63"}  name=runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d operation=remove
Sep 14 10:23:09 gitlab-runner-bastion gitlab-runner[18655]: ERROR:         * Resource not found: [DELETE https://ecs.eu-de.otc.t-systems.com/v2.1/f63d1faef3f747e9aa5cbcf017252ed1/servers/], error message: {"message":"API not found","request_id":"3ff1331e9c21dec814772ea059475c63"}  name=runner-kutriyer-gitlab-runner-autoscale-1600077912-5cf6976d operation=remove


```
Interestingly the Error varies a little bit from time to time

In the OTC Console the VM in question clearly exists and is running.

The Problem with this is, that after time passes the autscaling won't work anymore because of the "dead" vms:

```
root@gitlab-runner-bastion:~# docker-machine ls
NAME                                                          ACTIVE   DRIVER   STATE     URL                         SWARM   DOCKER      ERRORS
runner-kutriyer-gitlab-runner-autoscale-1600064125-b4fb3502            otc      Timeout
runner-kutriyer-gitlab-runner-autoscale-1600066625-90ffaa89   -        otc      Error                                         Unknown     IP address is not set
runner-kutriyer-gitlab-runner-autoscale-1600067580-8839c5af   -        otc      Error                                         Unknown     IP address is not set
runner-kutriyer-gitlab-runner-autoscale-1600071180-29dba128            otc      Timeout
runner-kutriyer-gitlab-runner-autoscale-1600071180-c480496b   -        otc      Running   tcp://192.168.67.227:2376           v19.03.12
runner-kutriyer-gitlab-runner-autoscale-1600071260-1f891f34   -        otc      Running   tcp://192.168.67.208:2376           v19.03.12
runner-kutriyer-gitlab-runner-autoscale-1600071440-e49a30c1   -        otc      Running   tcp://192.168.67.235:2376           v19.03.12
root@gitlab-runner-bastion:~# docker-machine ls
NAME                                                          ACTIVE   DRIVER   STATE     URL                         SWARM   DOCKER      ERRORS
runner-kutriyer-gitlab-runner-autoscale-1600064125-b4fb3502            otc      Timeout
runner-kutriyer-gitlab-runner-autoscale-1600066625-90ffaa89   -        otc      Error                                         Unknown     IP address is not set
runner-kutriyer-gitlab-runner-autoscale-1600067580-8839c5af   -        otc      Error                                         Unknown     IP address is not set
runner-kutriyer-gitlab-runner-autoscale-1600071900-8b6d5367   -        otc      Error                                         Unknown     IP address is not set
runner-kutriyer-gitlab-runner-autoscale-1600073280-c872dfd3   -        otc      Error                                         Unknown     IP address is not set
runner-kutriyer-gitlab-runner-autoscale-1600074100-87db8558   -        otc      Running   tcp://192.168.67.29:2376            v19.03.12
runner-kutriyer-gitlab-runner-autoscale-1600074110-b768ba75   -        otc      Running   tcp://192.168.67.7:2376             v19.03.12
runner-kutriyer-gitlab-runner-autoscale-1600074350-20c17883   -        otc      Running   tcp://192.168.67.169:2376           v19.03.12
root@gitlab-runner-bastion:~# docker-machine ls
NAME                                                          ACTIVE   DRIVER   STATE     URL                         SWARM   DOCKER      ERRORS
runner-kutriyer-gitlab-runner-autoscale-1600064125-b4fb3502            otc      Timeout
runner-kutriyer-gitlab-runner-autoscale-1600066625-90ffaa89   -        otc      Error                                         Unknown     IP address is not set
runner-kutriyer-gitlab-runner-autoscale-1600067580-8839c5af   -        otc      Error                                         Unknown     IP address is not set
runner-kutriyer-gitlab-runner-autoscale-1600071900-8b6d5367   -        otc      Error                                         Unknown     IP address is not set
runner-kutriyer-gitlab-runner-autoscale-1600073280-c872dfd3   -        otc      Error                                         Unknown     IP address is not set
runner-kutriyer-gitlab-runner-autoscale-1600074350-20c17883   -        otc      Running   tcp://192.168.67.169:2376           v19.03.12
runner-kutriyer-gitlab-runner-autoscale-1600074540-87d7b238            otc      Timeout
runner-kutriyer-gitlab-runner-autoscale-1600074550-2cf7c492   -        otc      Error                                         Unknown     IP address is not set
```

The Options set for docker-machine are:

```
  [runners.machine]
    IdleCount = 3
    MaxBuilds = 1
    MachineDriver = "otc"
    MachineName = "gitlab-runner-autoscale-%s"
    MachineOptions = ["otc-access-key-id=xxxxx", "otc-access-key-key=xxxxx", "otc-availability-zone=eu-de-03", "otc-flavor-name=s3.medium.4", "otc-image-id=455332f1-9bab-4628-b6ce-86cf721d467f", "otc-region=eu-de", "otc-sec-groups=gitlab-runners", "otc-ssh-user=ubuntu", "otc-subnet-id=671ecce8-xxxx-4ff8-910e-8c362bd0892b", "otc-tenant-id=xxxxx", "otc-vpc-id=bba043ce-cada-4537-8ea0-5475e9080c30", "otc-skip-default-sg", "otc-skip-ip", "otc-keypair-name=KeyPair-gitlab-runner-autoscale", "otc-private-key-file=/root/KeyPair-gitlab-runner-autoscale"]
```
This doesn't happen when calling docker-machine create manually with the same options.

This also doesn't happen with the old dockermachinedriver4otc driver in combination with gitlab runner

It happens at after around half an hour. The Driver is certainly able to create and remove machines with gitlab-runner just fine but after a while it goes wrong.

**Comment by AdamVB** [comment]
Just to add one more thing,

instance ID is empty in the json file of the affected machines

```

/root/.docker/machine/machines/runner-kutriyer-gitlab-runner-autoscale-1600079302-75fb22e8/config.json

        },
        "endpoint_type": "public",
        "instance_id": "",
        "key_pair": {
            "value": "KeyPair-gitlab-runner-autoscale",
            "managed": false
        },
```

**Comment by outcatcher** [comment]
As discussed: the problem was `s3...` flavors were faulty, machines created using `compute/v2` API not destructing after errors during creation. Should be fixed by #51
